h2o-genmodel/src/main/java/hex/genmodel/algos/targetencoder/TargetEncoderMojoModel.java

```diff
@@ -0,0 +1,123 @@
+package hex.genmodel.algos.targetencoder;
+
+import hex.genmodel.MojoModel;
+
+import java.util.*;
+
+public class TargetEncoderMojoModel extends MojoModel {
+
+  public TargetEncoderMojoModel(String[] columns, String[][] domains, String responseName) {
+    super(columns, domains, responseName);
+  }
+
+  public EncodingMaps _targetEncodingMap;
+  public Map<String, Integer> _teColumnNameToIdx = new HashMap<>();
+  public boolean _withBlending;
+  public double _inflectionPoint;
+  public double _smoothing;
+
+  // Could be passed from the model training phase so that we don't have to recompute it here. Or maybe it is fine as it should be computed only once per mojoModel
+  private double _priorMean = -1;
```

Comment:
i would just calculate this before the mojo is even created and store it explicitly - recalculating is not necessary

Reply:
@michalkurka I believe that there is no REcalculation. It will be done once. As my comment says it could be done `before` mojo. But this is mostly about some duplication of code/logic that computes mean. Will try to optimise it as well but I think it should not block this PR from merging.